src/lib/seam/thermostats/use-update-off.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  useMutation,
+  type UseMutationResult,
+  useQueryClient,
+} from '@tanstack/react-query'
+import type {
+  SeamError,
+  ThermostatDevice,
+  ThermostatOffRequest,
+  ThermostatsListResponse,
+} from 'seamapi'
+
+import { NullSeamClientError, useSeamClient } from 'lib/seam/use-seam-client.js'
+
+// UPSTREAM: Missing ThermostatOffResponse in seamapi.
+type UseSetThermostatOffData = Record<string, unknown>
+type UseSetThermostatOffMutationParams = ThermostatOffRequest
+
+export function useSetThermostatOff(): UseMutationResult<
+  UseSetThermostatOffData,
+  SeamError,
+  UseSetThermostatOffMutationParams
+> {
+  const { client } = useSeamClient()
+  const queryClient = useQueryClient()
+
+  return useMutation<
+    UseSetThermostatOffData,
+    SeamError,
+    UseSetThermostatOffMutationParams
+  >({
+    mutationFn: async (mutationParams: UseSetThermostatOffMutationParams) => {
+      if (client === null) throw new NullSeamClientError()
+
+      return await client.thermostats.off(mutationParams)
+    },
+    onSuccess: (_data, variables) => {
+      queryClient.setQueryData<ThermostatDevice | null>(
+        ['devices', 'get', { device_id: variables.device_id }],
+        (thermostat) => {
+          if (thermostat == null) {
+            return
+          }
+
+          return getUpdatedThermostat(thermostat)
+        }
+      )
+
+      queryClient.setQueryData<ThermostatsListResponse['thermostats']>(
+        ['devices', 'list', { device_id: variables.device_id }],
+        (thermostats): ThermostatDevice[] => {
+          if (thermostats == null) {
+            return []
+          }
+
+          return thermostats.map((thermostat) => {
+            if (thermostat.device_id === variables.device_id) {
+              return getUpdatedThermostat(thermostat)
+            }
+
+            return thermostat
+          })
+        }
+      )
+    },
+  })
+}
+
+function getUpdatedThermostat(thermostat: ThermostatDevice): ThermostatDevice {
+  return {
+    ...thermostat,
+    properties: {
+      ...thermostat.properties,
+      is_fan_running: false,
```

Comment:
Are we sure this always results in the fan turning off?

<img width="325" alt="image" src="https://github.com/seamapi/react/assets/721372/f8b12832-4e34-4a60-8dc9-ec2fa04b5821">